modules/tensor_mechanics/src/materials/Compute2DSmallStrain.C

```diff
@@ -17,25 +17,74 @@ validParams<Compute2DSmallStrain>()
 {
   InputParameters params = validParams<ComputeSmallStrain>();
   params.addClassDescription("Compute a small strain in a plane strain configuration.");
+
+  MooseEnum outOfPlaneDirection("x y z", "z");
+  params.addParam<MooseEnum>(
+      "out_of_plane_direction", outOfPlaneDirection, "The direction of the out-of-plane strain.");
   return params;
 }
 
 Compute2DSmallStrain::Compute2DSmallStrain(const InputParameters & parameters)
-  : ComputeSmallStrain(parameters)
+  : ComputeSmallStrain(parameters),
+    _out_of_plane_direction(getParam<MooseEnum>("out_of_plane_direction"))
 {
 }
 
+void
+Compute2DSmallStrain::initialSetup()
+{
+  displacementIntegrityCheck();
+  for (unsigned int i = 0; i < 3; ++i)
+  {
+    if (_out_of_plane_direction == i)
+    {
+      _disp[i] = &_zero;
+      _grad_disp[i] = &_grad_zero;
+    }
+    else
+    {
+      _disp[i] = &coupledValue("displacements", i);
+      _grad_disp[i] = &coupledGradient("displacements", i);
+    }
+  }
+}
+
 void
 Compute2DSmallStrain::computeProperties()
 {
   Real volumetric_strain = 0.0;
   for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
   {
-    _total_strain[_qp](0, 0) = (*_grad_disp[0])[_qp](0);
-    _total_strain[_qp](1, 1) = (*_grad_disp[1])[_qp](1);
-    _total_strain[_qp](0, 1) = ((*_grad_disp[0])[_qp](1) + (*_grad_disp[1])[_qp](0)) / 2.0;
-    _total_strain[_qp](1, 0) = _total_strain[_qp](0, 1); // force the symmetrical strain tensor
-    _total_strain[_qp](2, 2) = computeStrainZZ();
+    switch (_out_of_plane_direction)
+    {
+      case 0: // x
```

Comment:
If you had a C++ enum, you'd actually be able to just switch on X, Y, and Z instead of commenting like this.